### ISSUE TITLE
When making voice or video call via bluetooth headset can hear party, but they can't hear me

### ISSUE THREAD

**Issue by rsramkis** [signalapp/Signal-Android#13782]
### Guidelines

- [X] I have searched searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] This isn't a feature request or a discussion topic

### Bug description

**Problem:**

When making voice or video call via Bluetooth headset (Edifier) I can hear and see the person on the call, but they can't hear me. 

**Troubleshoot:**

1. Tried Discord voice calls with Edifier Bluetooth headset on same phone and the person receiving the call and can hear me.

2. Made voice call on Signal (no headset) and person heard me off built in microphone.

**Re-produce:**

1. Connect Bluetooth headset to phone wireless.
2. Call person on your your call list.

**Expected Result:**

1. The receiver of my call should hear me.
2. I should hear the receiver of the call.

Actual Results:

1. I hear the receiver of the call. But they cannot hear me.

### Device

Mororola Moto Z3 Play

### Android version

Lineage OS 21-20241023-NIGHTLY-beckham (Android 14)

### Signal version

7.22.2

### Link to debug log

https://debuglogs.org/android/7.22.2/8af9c9d7ea5e62b6beef0a3cffaee68d9844e92d889c21fa96a3e19e7304dce4

**Comment by rsramkis** [comment]
More about the handset:
https://wiki.lineageos.org/devices/beckham/

**Comment by jim-signal** [comment]
@rsramkis Thank you for the report. We have tested a few different bluetooth devices with the LineageOS, but not this combination. Can you let us know which specific Edifier device you are using? We can also try some tests with the newer LineageOS image.

**Comment by atomspring** [comment]
I'm having this issue, whether on the built-in microphone or a Bluetooth headset. I cannot make/receive _any_ calls on signal, the other person cannot hear me although I can hear them. 

Curiously, switching to speakerphone works but they say I'm extremely faint.

**Comment by jim-signal** [comment]
@atomspring We would need a log file from your device in order to troubleshoot what might be going on. Could you provide one, perhaps right after a non-working call? [The instructions to submit logs are here.](https://support.signal.org/hc/en-us/articles/360007318591-Debug-Logs-and-Crash-Reports) Thanks!

**Comment by rsramkis** [comment]
> @rsramkis Thank you for the report. We have tested a few different bluetooth devices with the LineageOS, but not this combination. Can you let us know which specific Edifier device you are using? We can also try some tests with the newer LineageOS image.

Hi Jim,

I have the following information for you:

1. The Bluetooth headset I am using is a Edifier W830BT.

2. If I use speaker phone on the device the other person hears echos of themselves (they can hear me talking to them).

3. If I use handset mode on the phone, no problem hearing and talking through the device.

**Comment by rsramkis** [comment]
> I'm having this issue, whether on the built-in microphone or a Bluetooth headset. I cannot make/receive _any_ calls on signal, the other person cannot hear me although I can hear them.
> 
> Curiously, switching to speakerphone works but they say I'm extremely faint.

I think you should open a different ticket to troubleshoot your issue. Seems you have different symptoms from mine.

**Comment by atomspring** [comment]
Indeed, sorry for the distraction.

~A

-------- Original Message --------

**Comment by rsramkis** [comment]
@jim-signal 

Thought I should let you know that I also tried using a Jabra Elite Active 65T bluetooth headphones and the same problems occurred with the receiver not able to hear my voice.

**Comment by rsramkis** [comment]
Is there any other information I can provide to help this ticket?

I have upgraded my phones OS to "Lineage OS 21-20241113-NIGHTLY-beckham (Android 14)" and Signal to 7.26.1.  

Same problem present.

**Comment by jim-signal** [comment]
I could not reproduce the issue with my LineageOS and Bluetooth device, although they are different from yours. I'll see if we can set a flag so the Z3 uses a different audio system, this should be done by tomorrow. Then we can see if it helps in your case.

**Comment by jim-signal** [comment]
@rsramkis It looks like the setting is live. Can you try again and let us know how it goes? If there is still and issue, can you provide a new log file? Maybe it will show something...

**Comment by rsramkis** [comment]
Hi Jim,

I repeated the testing as follows where the receiver of the call uses Apple and Android devices::

1. The Apple user calls the Moto Z3 Play (Signal 7.26.1)  on their Apple iOS 18.1.1tablet (Signal Version 7.39):
- Now they can hear me when using Bluetooth, but I can't hear them.
- We can hear each other when I use the ear piece or speaker phone on the phone (no Bluetooth).

2. The Moto Z3 Play calls the Apple iOS 18.1.1 Tablet (Signal Version 7.39):
- Now they can hear me when using Bluetooth, but I can't hear them.
- We can hear each other when I use the ear piece or speaker phone on the phone (no Bluetooth).

3. I repeated tests 1 + 2 with the users iPhone iOS 18.1.1 (Signal Version 7.39) calling the Moto Z3 Play and the same results.

4. The Moto Z3 Play calling a Moto G5 2024
- The test user can have full conversation with me on the Moto Z3 Play while I use Bluetooth headset (Edifier).
- The test user can have full conversation with me on the Moto Z3 Play while I use speaker phone.
- The test user can have full conversation with me on the Moto Z3 Play while I use ear piece.

**In summary:**

- The symptom changed so now the Apple user can hear me. But I can't hear them while using Bluetooth.
- New Android Test User (Moto G5 2024 with genuine Android 13) has no issues communicating with the Moto Z3 Play.

**Moto Z3 Play Debug log:**
https://debuglogs.org/android/7.26.1/d0680056ff6248c9232f06d8a75df3f8e2aa2d53cf791f28a57776d0547cf745

**Apple Tablet Debug log:**
https://debuglogs.org/ios/7.39.0/31afc226c49e821ec2bf6b32fca030cba0901f566a751a924f7e842cbc20a41d.zip

**Comment by jim-signal** [comment]
@rsramkis Thanks for all the debugging. Unfortunately, so far I am not seeing a reason in the logs why you can't hear the tablet in cases 1, 2, and 3. From the logs, it looks like audio data is sometimes being exchanged and has some energy. I am trying to see if I have another device I can try that is closer to yours.

**Comment by rsramkis** [comment]
> @rsramkis Thanks for all the debugging. Unfortunately, so far I am not seeing a reason in the logs why you can't hear the tablet in cases 1, 2, and 3. From the logs, it looks like audio data is sometimes being exchanged and has some energy. I am trying to see if I have another device I can try that is closer to yours.

I ended up doing more testing over the holidays. And it looks like even the Moto Z3 Play calling a Moto G5 2024 the sound will not be heard on a bluetooth device. You can switch to headset for speakerphone and both parties hear each other.

The versions I tested on Signal for Android = 7.28.4 and LineageOS 21 Android 14 December 25, 2024 build.

**Comment by rsramkis** [comment]
Thought I would add an update that the Moto G5 2024 has a physically plugged in headset and I can hear them on my computer or phone (but they can't hear me). So we are back to only talking on the the built in phone speaker or speaker phone. They use the same plugged in headset on regular cellular phone calls with no issues.

Seeing several tickets files with audio issues I can only assume Signal has a bigger problem.

**Comment by jim-signal** [comment]
@rsramkis For many audio issues, we need to be able to reproduce and verify a fix. I don't have access to a Moto G5, but I'll try to get as close as possible. Can you send a log from the Moto G5 device that captures a problematic call? We can look further.

**Comment by rsramkis** [comment]
Hi Jim,

Thanks for your feedback. What I was thinking is it is not realistic for a development team to have access to the exact device to reproduce (as I face the same challenge as a QA). That being said this issue has been open since November 2024. Seems like we need to take a different approach to resolving it. 

I really only have access to the Motorola Moto Z3 Play to help with the issue. It is very inconvenient that I can't use a bluetooth headset with this application anymore. As I mention ... Discord does work on the same device.

So wonder if we can reach out to your development team to see what other options we could use if the current debug logs are not sufficient. I can always provide you with another set.

**Comment by rsramkis** [comment]
Android Phone updated to Lineage OS 21-20241225-NIGHTLY-beckham (final version)

**I repeated testing after many Signal software updates:**

1. The Apple user calls the Moto Z3 Play (using bluetooth headset) on their Apple iOS 18.4.1 phone:

- Apple user hears Android user.
- Android does not hear the Apple user through headset.
- Switching Android to speaker allows Android user to hear Apple user.

2. The Moto Z3 Play (using bluetooth headset) calls the Apple Phone:
 
- Apple user hears Android user.
- Android hears the Apple user.
- Switching Android to speaker also allows both parties to hear each other.

The testing includes making both Audio and Video calls over 3 phone sessions included in the below debug logs.

Android Signal 7.39.5 Debug logs:
https://debuglogs.org/android/7.39.5/71ac500a8ccf0a83159723b0cfa6fa80578398d3f359129091070775c710c2df

Apple Signal 7.56 (747) Debug logs:
https://debuglogs.org/ios/7.56.0/b3f5f07c6315eef8dbe3d4611a382b4e1479a5dc385bcaf2218cc4c9ac8921a9.zip

**Comment by marek-jakub** [comment]
I can confirm the same issue using Moto g84, with Android 14, API 34 and Signal 7.41.3. Connected bluetooth speaker, the other party cannot hear me. Disconnecting the bluetooth speaker and using phone speaker solves the mic issue.

Please, find the debug log at:
https://debuglogs.org/android/7.41.3/2f2c17c2cda527c7d73f2944ece10b1fd1b5823d394256c0dbb127b970d11659

**Comment by rsramkis** [comment]
After moving to Android Signal version 7.72.2 Bluetooth headset is now working for calls. Closing the issue as fixed (with no root cause identified).

Note that Bluetooth devices in testing have not been upgraded or flashed.
